transcom/mymove
# B-20072

## [B-20072](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950860)
## [Integration Review](https://github.com/transcom/mymove/pull/13127)

## Summary

This task allows SC with the supervisor privilege (non-supervisor SC will retain the old filter) access to a typeahead multiselect for the origin duty location table. This dropdown contains all duty locations that are currently available in the queue they are viewing. 

I ended up having to use OrderFetcher to return the duty locations for the dropdowns due to the number of filters that are applied, and number of joins required, on the query to determine which moves are shown in a users queue. The other options was to recreate most of this query, and several associated functions, in a separate location_fetcher. After discussion with another dev it was decided that the amount of duplication required would be a worse option.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Prior to applying the Supervisor attribute in the user privileges table:

1. Access the Office application
2. Login as a Service Counselor
3. Verify you do not have access to the new Typeahead Multiselect filter for Origin Duty Location

After giving the user you plan to use for testing the Supervisor attribute in the user_privileges table:

SQL to add: 
INSERT INTO public.users_privileges VALUES (*your userId here*, '463c2034-d197-4d9a-897e-8bbe64893a31', '1d32fa97-b103-4386-9fec-2b3282ceb1be', '2018-03-27 15:01:23.020914', '2018-03-27 15:01:23.020914', null);

1. Access the Office application
2. Login as a Service Counselor
3. Select an option from the Origin Duty Location filter
4. Select another option from the Origin Duty Location filter
6. Type into the Origin Duty Location filter and verify typeahead is working

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
